packages/wallet/src/KeyManagement/types.ts

```diff
@@ -151,8 +158,20 @@ export interface KeyAgent {
    * @throws AuthenticationError
    */
   exportRootPrivateKey(): Promise<Cardano.Bip32PrivateKey>;
+  /**
+   * @throws AuthenticationError
+   */
+  signVotingMetadata(props: SignVotingMetadataProps): Promise<Cardano.Ed25519Signature>;
```

Comment:
`KeyAgent` interface is currently modeled to have a static `purpose`, `keyType` and `account` which doesn't match the up-and-coming [voting cip proposal](https://github.com/input-output-hk/gov-delegation-tools/blob/main/src/voting/CIP-voting.md) (it expects to wallet to use a different `purpose` etc.).

I suggest removing this and:
- extend `SerializableInMemoryKeyAgentData` to support using different `purpose` and `keyType`
- add util to create voting-specific `SerializableInMemoryKeyAgentData` (takes in `encryptedRootPrivateKeyBytes` and `networkId`), so it can be derived from the same root private key if desired.
- add util to `signVotingMetadata(keyAgent: KeyAgent, props: SignVotingMetadataProps)` (separate from KeyAgent interface or any specific key agent implementation) that uses `KeyAgent.signBlob` to do the actual signing
- add optional `purpose`, `keyType` and `accountIndex` in `KeyAgent.signBlob` params - it could be used for validating that correct instance of InMemoryKeyAgent is being used to sign voting metadata